frontend/packages/component-library/src/cms/heroBanner/HeroBanner.tsx

```diff
@@ -0,0 +1,111 @@
+  /** HeroBanner heading */
+  heading: string | ReactNode;
+  /** HeroBanner subHeading */
+  subHeading?: string | ReactNode;
+  /** HeroBanner description */
+  description?: string | ReactNode;
+  /** HeroBanner image */
+  imageProps?: ImageProps;
+  /** HeroBanner video */
+  videoProps?: VideoProps;
+  /** HeroBanner link */
+  buttonProps?: LinkButtonProps;
+  /** HeroBanner custom background url */
+  backgroundImageUrl?: string;
+  /** HeroBanner theme value.
+   *  If you're using backgroundImageUrl - you should make sure you set correct theme value to HeroBanner.
+   *  */
+  'data-theme'?: Theme;
+  /** Whether to show the background color */
+  removeBackground?: boolean;
+  /** HeroBanner whether show with wide text (text is wider than image) */
+  withWideText?: boolean;
+  /** HeroBanner partner prop */
+  partner?: {title: string; logo: ImageProps};
+  /** HeroBanner custom className  */
+  className?: string;
+}
+
+/**
+ * ## Production-ready Checklist:
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✔) has tests: test every prop, every state and every interaction that's js related;
+ *  * (see ./__tests__/HeroBanner.test.tsx)
+ *  * (✔) passes accessibility checks;
+ *
+ * ### Status: ```Ready for dev```
+ *
+ * Design System: HeroBanner Component.
+ * Renders a Hero Banner/Section which serves as an opening section of a page.
+ */
+const HeroBanner: React.FC<HeroBannerProps> = ({
+  heading,
+  subHeading,
+  description,
+  partner,
+  buttonProps,
+  imageProps,
+  videoProps,
+  backgroundImageUrl,
+  removeBackground = false,
+  withWideText = false,
+  className,
+  ...HTMLAttributes
+}) => (
+  <section
+    {...HTMLAttributes}
+    className={classNames(moduleStyles.heroBannerWrapper, className, {
+      [moduleStyles['heroBanner-withWideText']]: withWideText,
+    })}
+    data-theme={HTMLAttributes['data-theme']}
+    style={{
+      ...(HTMLAttributes.style ?? {}),
+      backgroundImage: backgroundImageUrl
+        ? `url(${backgroundImageUrl})`
+        : HTMLAttributes.style?.backgroundImage,
+      ...(removeBackground ? {background: 'none'} : {}),
+    }}
+  >
+    <div className={classNames(moduleStyles.heroBannerContainer)}>
+      <div className={moduleStyles.heroBannerDescriptiveSection}>
+        <Heading1>{heading}</Heading1>
+
+        {subHeading && <BodyOneText>{subHeading}</BodyOneText>}
+
+        {description && <BodyTwoText>{description}</BodyTwoText>}
+
+        {partner && (
+          <span className={moduleStyles.heroBannerPartnerSection}>
+            {partner.title}
+            <Image {...partner.logo} />
+          </span>
+        )}
+
+        {buttonProps && (
+          <LinkButton color="purple" type="primary" {...buttonProps} />
+        )}
+      </div>
+      {(imageProps || videoProps) && (
+        <div className={moduleStyles.heroBannerVisualSection}>
```

Comment:
Another naming super nit:
```suggestion
        <div className={moduleStyles.heroBannerMediaWrapper}>
```